engine/src/main/java/org/terasology/engine/TerasologyEngine.java

```diff
@@ -501,31 +529,24 @@ private void mainLoop() {
         }
         PerformanceMonitor.endActivity();
 
-        // This becomes important only if display.isCloseRequested() is true.
-        // In all other circumstances the EngineState is already set to
-        // INITIALIZED by the time the flow gets here.
-        engineState = EngineState.INITIALIZED;
+        changeStatus(StandardGameStatus.SHUTTING_DOWN);
     }
 
     private void cleanup() {
         logger.info("Shutting down Terasology...");
+        shuttingDown = true;
+        changeStatus(StandardGameStatus.SHUTTING_DOWN);
```

Comment:
It seems to me that this line is perhaps redundant with the identical line at the end of mainLoop().